src/ucs/arch/aarch64/cpu.h

```diff
@@ -94,6 +95,33 @@ static inline void ucs_arch_wait_mem(void *address)
                   : "Q"(address));
 }
 
+static inline void ucs_clear_cache(void *start, void *end)
+{
+#if HAVE___CLEAR_CACHE
+    /* do not allow global declaration of compiler intrinsic */
+    void __clear_cache(void* beg, void* end);
+
+    __clear_cache(start, end);
+#elif HAVE___AARCH64_SYNC_CACHE_RANGE
+    void __aarch64_sync_cache_range(void* beg, void* end);
+
+    __aarch64_sync_cache_range(start, end);
+#else
+    uintptr_t sptr = ucs_align_down((uintptr_t)start, UCS_ARCH_CACHE_LINE_SIZE);
+    uintptr_t ptr;
+
+    for (ptr = sptr; ptr < (uintptr_t)end; ptr += UCS_ARCH_CACHE_LINE_SIZE) {
+        asm volatile ("dc cvau, %0" :: "r" (ptr) : "memory");
+    }
+    asm volatile ("dsb ish" ::: "memory");
+
+    for (ptr = sptr; ptr < (uintptr_t)end; ptr += UCS_ARCH_CACHE_LINE_SIZE) {
+        asm volatile ("ic ivau, %0" :: "r" (ptr) : "memory");
+    }
+    asm volatile ("dsb ish; isb" ::: "memory");
```

Comment:
Where the code is coming from ? I don't think that dsb following by isb makes much sense.

Reply:
https://github.com/gcc-mirror/gcc/blob/577b0b07be7b7ae4950e28f4221b04adbe123d74/libgcc/config/aarch64/sync-cache.c#L71

Reply:
@shamisp ?

Reply:
The code is fine. You have to use ISB since DSB only order instructions that read/store memory. If you have jump and other you have to add ISB.

Reply:
@shamisp looks like the code is doing both, no?

Reply:
@yosefe correct. I'm only not sure about the cache line size. I think for this particular case it might be safer to read it from register instead of using configuration option.

Reply:
@yosefe ideas?
I can add reading cache line from processor